packages/backend/src/peripherals/ethereum/MulticallClient.ts

```diff
@@ -1,30 +1,16 @@
-import { Bytes, EthereumAddress } from '@l2beat/shared-pure'
-import { utils } from 'ethers'
-
 import { EthereumClient } from './EthereumClient'
-
-export const MULTICALL_BATCH_SIZE = 150
-export const MULTICALL_V1_BLOCK = 7929876
-export const MULTICALL_V1_ADDRESS = EthereumAddress(
-  '0xeefBa1e63905eF1D7ACbA5a8513c70307C1cE441',
-)
-export const MULTICALL_V2_BLOCK = 12336033
-export const MULTICALL_V2_ADDRESS = EthereumAddress(
-  '0x5BA1e12693Dc8F9c48aAD8770482f4739bEeD696',
-)
-
-export interface MulticallRequest {
-  address: EthereumAddress
-  data: Bytes
-}
-
-export interface MulticallResponse {
-  success: boolean
-  data: Bytes
-}
+import { ETHEREUM_MULTICALL_CONFIG } from './MulticallConfig'
+import {
+  MulticallConfigEntry,
+  MulticallRequest,
+  MulticallResponse,
+} from './types'
 
 export class MulticallClient {
-  constructor(private readonly ethereumClient: EthereumClient) {}
+  constructor(
+    private readonly ethereumClient: EthereumClient,
+    private readonly config = ETHEREUM_MULTICALL_CONFIG,
```

Comment:
I do not like this default param, maybe a static method?

Reply:
Removed